src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt

```diff
@@ -1,80 +1,80 @@
-  @Autowired(required = false)
-  private val buildProperties: BuildProperties? = null
+    /**
+     * @return health data. Note this is unsecured so no sensitive data allowed!
+     */
+    private val version: String
+        get() = if (buildProperties == null) "version not available" else buildProperties.version
 
-  /**
-   * @return health data. Note this is unsecured so no sensitive data allowed!
-   */
-  private val version: String
-    get() = if (buildProperties == null) "version not available" else buildProperties.version
+    @Throws(Exception::class)
+    override fun configure(http: HttpSecurity) {
 
-  @Throws(Exception::class)
-  override fun configure(http: HttpSecurity) {
+        http.headers().frameOptions().sameOrigin().and()
+                .sessionManagement()
+                .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
 
-    http.headers().frameOptions().sameOrigin().and()
-        .sessionManagement()
-        .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                // Can't have CSRF protection as requires session
+                .and().csrf().disable()
+                .authorizeRequests { auth ->
+                    auth.antMatchers(
+                                    "/webjars/**", "/favicon.ico", "/csrf",
+                                    "/health", "/info", "/ping", "/h2-console/**",
+                                    "/v2/api-docs",
+                                    "/swagger-ui.html", "/swagger-resources", "/swagger-resources/configuration/ui",
+                                    "/swagger-resources/configuration/security"
+                            ).permitAll()
+                            .anyRequest()
+                            .authenticated()
+                }.oauth2ResourceServer()
+                .jwt()
+    }
 
-        // Can't have CSRF protection as requires session
-        .and().csrf().disable()
-        .authorizeRequests()
-        .antMatchers("/webjars/**", "/favicon.ico", "/csrf",
-            "/health", "/info", "/health/ping").permitAll()
-        .anyRequest()
-        .authenticated()
-  }
-
-  override fun configure(config: ResourceServerSecurityConfigurer) {
-    config.tokenServices(tokenServices())
-  }
-
-  @Bean
-  open fun tokenStore(): TokenStore = JwtTokenStore(accessTokenConverter())
-
-  @Bean
-  open fun accessTokenConverter(): JwtAccessTokenConverter {
-    val converter = JwtAccessTokenConverter()
-    converter.setVerifierKey(String(Base64.decodeBase64(jwtPublicKey)))
-    return converter
-  }
-
-  @Bean
-  @Primary
-  open fun tokenServices(): DefaultTokenServices {
-    val defaultTokenServices = DefaultTokenServices()
-    defaultTokenServices.setTokenStore(tokenStore())
-    return defaultTokenServices
-  }
-
-  @Bean
-  @ConfigurationProperties("prisontonhs.client")
-  open fun prisonToNhsClientCredentials(): ClientCredentialsResourceDetails = ClientCredentialsResourceDetails()
+    @Bean
+    open fun api(): Docket? {
```

Comment:
I think a lot of place like this returns nullable unnecessary?

Reply:
changed